### PR TITLE
Add endpoints for fetching overall insights and logs within a time interval.

### PR DESCRIPTION
**Pull request by AmartC**
This PR adds support for fetching logs within a start and end timestamp by implementing a logs endpoint for the HTTP server. It also adds support for fetching overall insights within a time interval by implementing a overall_insights endpoint for the HTTP server.

### CODE REVIEW

File: opni-insights-service/app/main.py [hunk 1]
```diff
@@ -307,3 +372,29 @@ async def index_workload(start_ts: int, end_ts: int):
...
+        return result
+    except Exception as e:
+        # Bad Request
+        logging.error(e)
+
+
+@app.get("/logs")
+async def index_logs(start_ts: int, end_ts: int):
+    # This function handles get requests for fetching workload breakdown insights.
+    logging.info(f"Received request to obtain all logs between {start_ts} and {end_ts}")
```
Comment (tybalex) [inline opni-insights-service/app/main.py:394]: the logging here says it requests for all logs, however this function `get_logs` returns non-normal logs?

File: opni-insights-service/app/main.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 # Third Party
 from elasticsearch import Elasticsearch
```
Comment (tybalex) [inline opni-insights-service/app/main.py:5]: Shouldn't we use AsyncElasticsearch?

**Review by tybalex — CHANGES_REQUESTED** [review]

File: opni-insights-service/app/main.py (hunk 2 above)
Comment (AmartC) [reply opni-insights-service/app/main.py:5]: I think using just Elasticsearch is okay for this.

File: opni-insights-service/app/main.py (hunk 1 above)
Comment (tybalex) [reply opni-insights-service/app/main.py:394]: @AmartC ^

File: opni-insights-service/app/main.py (hunk 2 above)
Comment (tybalex) [reply opni-insights-service/app/main.py:5]: for example: `es_instance.count()` is called in `get_overall_breakdown()` which is used by `async def index_overall_breakdown(start_ts: int, end_ts: int):`.
So we should either use AsyncES, or define `index_overall_breakdown()` as non-async function?

**Review by tybalex — DISMISSED** [review]

**Review by tybalex — APPROVED** [review]

File: opni-insights-service/app/main.py [hunk 3]
```diff
@@ -267,14 +272,75 @@ def get_namespace_breakdown(start_ts, end_ts):
...
+            }
+        }
+        try:
+            overall_breakdown_dict[anomaly_level] = (
+                await es_instance.count(index="logs", body=query_body)
+            )["count"]
+        except Exception as e:
+            logging.error(f"Unable to access Elasticsearch data. {e}")
+            return overall_breakdown_dict
+    return overall_breakdown_dict
```
Comment (dbason) [inline opni-insights-service/app/main.py:298]: This is fine for now, but we should be using an aggregation in ES rather than a loop with multiple queries.

**Review by dbason — APPROVED** [review]

**Pull request merged**